typescript/schedules.md

```diff
@@ -0,0 +1,161 @@
+# Schedules
+
+- [Docs](https://docs.temporal.io/workflows/#schedules)
+- gRPC API:
+  - [methods](https://github.com/temporalio/api/blob/799926c86eb13d8a9717d3561ab9b0df43796c06/temporal/api/workflowservice/v1/service.proto#L328-L370)
+  - [`request_response.proto`](https://github.com/temporalio/api/blob/799926c86eb13d8a9717d3561ab9b0df43796c06/temporal/api/workflowservice/v1/request_response.proto#L821-L957)
+  - [`schedule/v1/message.proto`](https://github.com/temporalio/api/blob/master/temporal/api/schedule/v1/message.proto)
+  - [`ScheduleOverlapPolicy` enum](https://github.com/temporalio/api/blob/master/temporal/api/enums/v1/schedule.proto)
+
+## TS API
+
+```ts
+import { ScheduleClient, ScheduleOverlapPolicy } from '@temporalio/client'
+import { addWeeks } from 'date-fns'
+import { myWorkflow } from './workflows'
+
+const client = new ScheduleClient()
+
+const schedule = await client.create({
+  id: 'schedule-biz-id',
+  spec: {
+    // every hour at minute 5
+    interval: {
+      every: '1h',
+      at: '5m',
+    },
+    // every 20 days since epoch at day 2
+    // interval: {
+    //   every: '20d',
+    //   at: '2d'
+    // }
+    exclude: {
+      // skip 11:05 pm
+      hour: 23,
+      minute: 5,
+    },
+    endAt: addWeeks(new Date(), 4), 
+    jitter: '30s',
+    timezone: 'US/Eastern',
```

Comment:
One of the features that was specially requested for timezone handling was the option to copy the actual tzinfo file from the client into the schedule, so the client always knows exactly what they're getting and doesn't have to depend on someone keeping the server's tzdb up to date. Most users won't need this, but for those that do, we should offer it as an option. (Exposing it in tctl is on my long todo list.)

Reply:
cc @underrun , whose idea this was

The idea is the client should have full control over timezone conversions, so that a server upgrade can't suddenly shift the timezone definition. Of course in any reasonable deployment + timezone, definition changes are published at least six months ahead of time, and both server and client are updated regularly. The question is what happens in suboptimal situations when things break down. Maybe the team maintaining the server doesn't touch it for a year since it's working fine (we definitely hear from open source users on very old server versions). Assuming it's a container image, the tzdb won't be updated. Maybe a government pushes a time zone update on super short notice and the application team wants control over when exactly it gets applied, instead of leaving it to server operators to decide for multiple applications within the company. This is definitely more work for the client, which is why most people shouldn't do it, it's just an option for teams who know they want to do it.

Clock skew is important, but I think it's a little different: you can run ntpd in your vm and you're good to go indefinitely for all reasonable purposes. Probably most cloud setups will include this automatically. Updating tzdb on the server requires deploying a new image, which is a bigger deal. And if you don't want to upgrade server versions, you might have to build your own image with an old binary and new tzdb.

Reply:
> it's just an option for teams who know they want to do it.

Can we wait until asked for it before putting in the SDK? I think it's reasonable, whether using k8s scheduling, crond, or whatever to count on the place where the schedule is evaluated to make sure it can evaluate accurately. But if we can defer this functionality anyways, no worries.

Reply:
Making an analogy to the single machine environment, the philosophical question is whether you consider Temporal to be more akin to the OS kernel, which keeps track of time but leaves most application-level policy, like time zones, to userspace, or something like systemd, which encodes a lot of policy.

I think we can defer it, just want to make sure people know why it's there.

Reply:
:+1: If we can point to other scheduling engines or job servers/clusters or whatever for inspiration, it can sure help me understand what it might look like and help me justify its usefulness to others (I could of course probably point to many that don't do this, but I suspect the issue is decided)